lutron-aurora/src/init.lua

```diff
@@ -0,0 +1,105 @@
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+local Level = zcl_clusters.Level
+local ZigbeeDriver = require "st.zigbee"
+local capabilities = require "st.capabilities"
+local device_management = require "st.zigbee.device_management"
+local utils = require 'st.utils'
+
+local PowerConfiguration = zcl_clusters.PowerConfiguration
+local Groups = zcl_clusters.Groups
+
+local LAST_LEVEL_EVENT = "LAST_LEVEL_EVENT"
+local DEVICE_GROUP = "DEVICE_GROUP"
+--server: Basic, PowerConfiguration, Identify, TouchlinkCommissioning, 0xFC00 
+--client: Identify, Groups, OnOff, Level, OTAUpgrade, TouchlinkCommissioning
+
+local do_configure = function(self, device)
+  device:send(device_management.build_bind_request(device, PowerConfiguration.ID, self.environment_info.hub_zigbee_eui))
+  device:send(device_management.build_bind_request(device, Level.ID, self.environment_info.hub_zigbee_eui))
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:configure_reporting(device, 30, 21600, 1))
+end
+
+local function added_handler(self, device)
+  for _, component in pairs(device.profile.components) do
+    device:emit_component_event(component, capabilities.button.numberOfButtons({ value = 1 }))
+    device:emit_event(capabilities.button.supportedButtonValues({"pushed"}, {visibility = { displayed = false }}))
+  end
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:read(device))
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:read(device))
+  device:emit_event(capabilities.button.button.pushed({ state_change = false }))
+  device:emit_event(capabilities.switchLevel.level(0, { state_change = false }))
+end
+
+local battery_perc_attr_handler = function(_, device, value, _)
+  device:emit_event(capabilities.battery.battery(utils.clamp_value(value.value, 0, 100)))
+end
+
+local function level_event_handler(driver, device, cmd)
+  -- The device is essentially a stateless dimmer, it will always send events with a transition_time
+  -- of 7 for on/off events and the value will be either 255 or 0, the memory on the device
+  -- seems to reset after about 1 second so 0 is a bit rare. If the transition_time is 2 then
+  -- the event is a dimmer event, the value with either will be 2 for a reduction or > 3
+  -- for an increase. We are converting these dimmer events into individual steps based on the
+  -- device preference
+  local level_step = device.preferences.stepSize or 5
+  local value = cmd.body.zcl_body.level.value
+  local time = cmd.body.zcl_body.transition_time.value
+  if time == 7 then
+    device:emit_event(capabilities.button.button.pushed())
```

Comment:
For most buttons we explicitly set `stateChange = true` if you don't do this, if you press the button twice in a row the second press won't be treated as a state change, which means consumers (e.g. rules/automations) won't be informed of the event.